homeassistant/components/mqtt/debug_info.py

```diff
@@ -101,65 +102,73 @@ def add_subscription(hass, message_callback, subscription):
         entity_info["subscriptions"][subscription]["count"] += 1
 
 
-def remove_subscription(hass, message_callback, subscription):
+def remove_subscription(
+    hass: HomeAssistant,
+    message_callback: MessageCallbackType,
+    subscription: str,
+) -> None:
     """Remove debug data for subscription if it exists."""
+    debug_info_entities = get_mqtt_data(hass).debug_info_entities
+
     entity_id = getattr(message_callback, "__entity_id", None)
-    if entity_id and entity_id in hass.data[DATA_MQTT_DEBUG_INFO]["entities"]:
-        hass.data[DATA_MQTT_DEBUG_INFO]["entities"][entity_id]["subscriptions"][
-            subscription
-        ]["count"] -= 1
-        if not hass.data[DATA_MQTT_DEBUG_INFO]["entities"][entity_id]["subscriptions"][
-            subscription
-        ]["count"]:
-            hass.data[DATA_MQTT_DEBUG_INFO]["entities"][entity_id]["subscriptions"].pop(
-                subscription
-            )
-
-
-def add_entity_discovery_data(hass, discovery_data, entity_id):
+    if entity_id and entity_id in debug_info_entities:
```

Comment:
Same here, by moving out of the if, you are assigning a variable you might never need.
Maybe use a walrus?